pkg/builder/controller.go

```diff
@@ -272,17 +291,23 @@ func (blder *Builder) project(obj client.Object, proj objectProjection) (client.
 	}
 }
 
-func (blder *Builder) doWatch() error {
+func (blder *TypedBuilder[request]) doWatch() error {
 	// Reconcile type
 	if blder.forInput.object != nil {
 		obj, err := blder.project(blder.forInput.object, blder.forInput.objectProjection)
 		if err != nil {
 			return err
 		}
-		hdler := &handler.EnqueueRequestForObject{}
+
+		var hdler handler.TypedEventHandler[client.Object, request]
+		if reflect.TypeFor[request]() != reflect.TypeOf(reconcile.Request{}) {
+			return errors.New("For() is not supported for TypedBuilder")
```

Comment:
I think that's technically not correct. Only TypedBuilder which are for a type which is not reconcile.Request are not supported (the Builder itself is a supported TypedBuilder)

Reply:
We only run into this if if the request type is not `reconcile.Request` and I guess the underlying assumption was that people was use the `Builder` type when they want a `TypedBuilder[reconcile.Request]`

Reply:
That is actually a good idea, updated it

Reply:
Thx!